### ISSUE TITLE
[Bug]: result is not correct for operation[vector + vector]

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#12361]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):93d5333e2ea08527bfe29542d13daca14cfebd28
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

the same query for 2 times on a solid table data, but the result is different

mysql> select id,vecf32_3 + vecf32_3, vecf32_5 + vecf32_5 from vtab32;
+------+-------------------------------------------------------+--------------------------------+
| id   | vecf32_3 + vecf32_3                                   | vecf32_5 + vecf32_5            |
+------+-------------------------------------------------------+--------------------------------+
|    1 | NULL                                                  | NULL                           |
|    2 | [0.3452598, 6.5817714, 60.866188]                     | NULL                           |
|    3 | [17.121378, 13.580718, 1643.9556]                     | [8.560689, 6.790359, 821.9778] |
|    4 | [0.46323407, 23.498016, 563.923, 56.076736, 8732.958] | [8.560689, 6.790359, 821.9778] |
+------+-------------------------------------------------------+--------------------------------+
4 rows in set (0.00 sec)

mysql> select id,vecf32_3 + vecf32_3, vecf32_5 + vecf32_5 from vtab32;
+------+-----------------------------------+-------------------------------------------------------+
| id   | vecf32_3 + vecf32_3               | vecf32_5 + vecf32_5                                   |
+------+-----------------------------------+-------------------------------------------------------+
|    1 | NULL                              | NULL                                                  |
|    2 | [0.3452598, 6.5817714, 60.866188] | NULL                                                  |
|    3 | [17.121378, 13.580718, 1643.9556] | [17.121378, 13.580718, 1643.9556]                     |
|    4 | [8.560689, 6.790359, 821.9778]    | [0.46323407, 23.498016, 563.923, 56.076736, 8732.958] |
+------+-----------------------------------+-------------------------------------------------------+
4 rows in set (0.01 sec)

<img width="752" alt="image" src="https://github.com/matrixorigin/matrixone/assets/47846308/4a912c8b-8107-4e5c-82f3-a36b206487d7">

the table data:
mysql> select * from vtab32;
+------+------------------------------------+--------------------------------------------------------+
| id   | vecf32_3                           | vecf32_5                                               |
+------+------------------------------------+--------------------------------------------------------+
|    1 | NULL                               | NULL                                                   |
|    2 | [0.8166459, 0.66616553, 0.4886152] | NULL                                                   |
|    3 | [0.1726299, 3.2908857, 30.433094]  | [0.45052445, 2.1984527, 9.579752, 123.48039, 4635.894] |
|    4 | [8.560689, 6.790359, 821.9778]     | [0.46323407, 23.498016, 563.923, 56.076736, 8732.958]  |
+------+------------------------------------+--------------------------------------------------------+
4 rows in set (0.01 sec)

### Steps to Reproduce

```Markdown
create table vtab32(id int primary key auto_increment,`vecf32_3` vecf32(3),`vecf32_5` vecf32(5));
insert into vtab32(vecf32_3,vecf32_5) values("[0.8166459,NULL,0.4886152]",NULL);
insert into vtab32(vecf32_3,vecf32_5) values(NULL,NULL);
insert into vtab32(vecf32_3,vecf32_5) values("[0.8166459,0.66616553,0.4886152]",NULL);
insert into vtab32(vecf32_3,vecf32_5) values("0.1726299,3.29088557,30.4330937","0.1726299,3.29088557,30.4330937");
insert into vtab32(vecf32_3,vecf32_5) values ("[0.1726299,3.29088557,30.4330937]","[0.45052445,2.19845265,9.579752,123.48039162,4635.89423394]");
insert into vtab32(vecf32_3,vecf32_5) values ("[8.5606893,6.7903588,821.977768]","[0.46323407,23.49801546,563.9229458,56.07673508,8732.9583881]");

select * from vtab64 join vtab32 on vtab64.vecf32_3 = vtab32.vecf32_3;
```

**Comment by aressu1985** [comment]
@arjunsk 

other operations [ * / - ] also have this probelem
mysql> select id,vecf32_3 - vecf32_3, vecf32_5 - vecf32_5 from vtab32;
+------+-------------------------------------------------------+-----------------------------------+
| id   | vecf32_3 - vecf32_3                                   | vecf32_5 - vecf32_5               |
+------+-------------------------------------------------------+-----------------------------------+
|    1 | NULL                                                  | NULL                              |
|    2 | [0, 0, 0]                                             | NULL                              |
|    3 | [0, 0, 0]                                             | [0.1726299, 3.2908857, 30.433094] |
|    4 | [0.46323407, 23.498016, 563.923, 56.076736, 8732.958] | [8.560689, 6.790359, 821.9778]    |
+------+-------------------------------------------------------+-----------------------------------+
4 rows in set (0.02 sec)

mysql> select id,vecf32_3 * vecf32_3, vecf32_5 * vecf32_5 from vtab32;
+------+-------------------------------------------------------+-----------------------------------+
| id   | vecf32_3 * vecf32_3                                   | vecf32_5 * vecf32_5               |
+------+-------------------------------------------------------+-----------------------------------+
|    1 | NULL                                                  | NULL                              |
|    2 | [0.02980108, 10.829928, 926.1732]                     | NULL                              |
|    3 | [73.28539, 46.108974, 675647.5]                       | [0.1726299, 3.2908857, 30.433094] |
|    4 | [0.46323407, 23.498016, 563.923, 56.076736, 8732.958] | [8.560689, 6.790359, 821.9778]    |
+------+-------------------------------------------------------+-----------------------------------+
4 rows in set (0.01 sec)

mysql> select id,vecf32_3 / vecf32_3, vecf32_5 / vecf32_5 from vtab32;
+------+--------------------------------+-----------------------------------+
| id   | vecf32_3 / vecf32_3            | vecf32_5 / vecf32_5               |
+------+--------------------------------+-----------------------------------+
|    1 | NULL                           | NULL                              |
|    2 | [1, 1, 1]                      | NULL                              |
|    3 | [1, 1, 1]                      | [0.1726299, 3.2908857, 30.433094] |
|    4 | [8.560689, 6.790359, 821.9778] | [8.560689, 6.790359, 821.9778]    |
+------+--------------------------------+-----------------------------------+
4 rows in set (0.01 sec)

**Comment by arjunsk** [comment]
The errors seems to occur when the table contains NULLS.

```sql
create table tbl(id int primary key auto_increment,`a` vecf32(3),`b` vecf32(5));

-- inserts 1 (no nulls) // Valid
insert into tbl(a,b) values("[1,2,3]","[1,2,3,4,5]");
insert into tbl(a,b) values("[4,5,6]","[6,7,8,9,10]");
select id,a + a, b + b from tbl;
+------+-------------+----------------------+
| id   | a + a       | b + b                |
+------+-------------+----------------------+
|    1 | [2, 4, 6]   | [2, 4, 6, 8, 10]     |
|    2 | [8, 10, 12] | [12, 14, 16, 18, 20] |
+------+-------------+----------------------+
select id,a + a, b + b from tbl;
+------+-------------+----------------------+
| id   | a + a       | b + b                |
+------+-------------+----------------------+
|    1 | [2, 4, 6]   | [2, 4, 6, 8, 10]     |
|    2 | [8, 10, 12] | [12, 14, 16, 18, 20] |
+------+-------------+----------------------+

-- inserts 2 (both nulls) // Valid
insert into tbl(a,b) values(NULL, NULL);
select id,a + a, b + b from tbl;
+------+-------------+----------------------+
| id   | a + a       | b + b                |
+------+-------------+----------------------+
|    1 | [2, 4, 6]   | [2, 4, 6, 8, 10]     |
|    2 | [8, 10, 12] | [12, 14, 16, 18, 20] |
|    3 | NULL        | NULL                 |
+------+-------------+----------------------+
select id,a + a, b + b from tbl;
+------+-------------+----------------------+
| id   | a + a       | b + b                |
+------+-------------+----------------------+
|    1 | [2, 4, 6]   | [2, 4, 6, 8, 10]     |
|    2 | [8, 10, 12] | [12, 14, 16, 18, 20] |
|    3 | NULL        | NULL                 |
+------+-------------+----------------------+

-- inserts 3 (one null, and another non null) // Invalid
insert into tbl(a,b) values("[7,8,9]", NULL);
select id,a + a, b + b from tbl;
+------+-------------+----------------------+
| id   | a + a       | b + b                |
+------+-------------+----------------------+
|    1 | [2, 4, 6]   | [2, 4, 6, 8, 10]     |
|    2 | [8, 10, 12] | [12, 14, 16, 18, 20] |
|    3 | NULL        | NULL                 |
|    4 | []          | NULL                 | <-- bug
+------+-------------+----------------------+
select id,a + a, b + b from tbl;
+------+-------------+----------------------+
| id   | a + a       | b + b                |
+------+-------------+----------------------+
|    1 | [2, 4, 6]   | [2, 4, 6, 8, 10]     |
|    2 | [8, 10, 12] | [12, 14, 16, 18, 20] |
|    3 | NULL        | NULL                 |
|    4 | [7, 8, 9]   | NULL                 |
+------+-------------+----------------------+

-- Insert 4 (one null, another non null)     // Invalid
insert into tbl(a,b) values("[2,3,4]", NULL);
select id,a + a, b + b from tbl;
+------+-------------+----------------------+
| id   | a + a       | b + b                |
+------+-------------+----------------------+
|    5 | [4, 6, 8]   | NULL                 | <-- invalid
|    1 | [2, 4, 6]   | [2, 4, 6, 8, 10]     |
|    2 | [8, 10, 12] | [12, 14, 16, 18, 20] |
|    3 | NULL        | NULL                 |
|    4 | []          | NULL                 |
+------+-------------+----------------------+
select id,a + a, b + b from tbl;
+------+-------------+----------------------+
| id   | a + a       | b + b                |
+------+-------------+----------------------+
|    1 | [2, 4, 6]   | [2, 4, 6, 8, 10]     |
|    2 | [8, 10, 12] | [12, 14, 16, 18, 20] |
|    3 | NULL        | NULL                 |
|    4 | [4, 6, 8]   | NULL                 |
|    5 | []          | NULL                 | <-- invalid
+------+-------------+----------------------+
```

**Comment by arjunsk** [comment]
If a table containing a vector has null then we are having some strange issues for binary operators.

```mysql
create table tbl(id int primary key auto_increment,`a` vecf32(3),`b` vecf32(5));
insert into tbl(a,b) values(NULL,NULL);
insert into tbl(a,b) values("[1,2,3]",NULL);
insert into tbl(a,b) values("[4,5,6]","[1,2,3,4,5]");
insert into tbl(a,b) values("[7,8,9]","[2,2,3,4,5]");
select id,a + a, b + b from tbl;
+------+--------------+--------------+
| id   | a + a        | b + b        |
+------+--------------+--------------+
|    1 | NULL         | NULL         |
|    2 | [8, 10, 12]  | NULL         |
|    3 | [14, 16, 18] | [14, 16, 18] |
|    4 | []           | []           |
+------+--------------+--------------+
delete from tbl where id=1;
delete from tbl where id=2;
select id,a + a, b + b from tbl;
+------+--------------+------------------+
| id   | a + a        | b + b            |
+------+--------------+------------------+
|    3 | [8, 10, 12]  | [2, 4, 6, 8, 10] |
|    4 | [14, 16, 18] | [4, 4, 6, 8, 10] |
+------+--------------+------------------+
```

**Comment by arjunsk** [comment]
This PR should fix the bug: https://github.com/matrixorigin/matrixone/pull/12369. Thanks to @m-schen for the support.

**Comment by arjunsk** [comment]
The PR is merged with the master. kindly retest.

**Comment by aressu1985** [comment]
testing

**Comment by aressu1985** [comment]
fixed commit: 4e4d1cd448867ac0fba7afcf53b576f72cd5ff30


mysql> select * from vtab32;
+------+------------------------------------+--------------------------------------------------------+
| id   | vecf32_3                           | vecf32_5                                               |
+------+------------------------------------+--------------------------------------------------------+
|    1 | NULL                               | NULL                                                   |
|    2 | [0.8166459, 0.66616553, 0.4886152] | NULL                                                   |
|    3 | [0.1726299, 3.2908857, 30.433094]  | [0.45052445, 2.1984527, 9.579752, 123.48039, 4635.894] |
|    4 | [8.560689, 6.790359, 821.9778]     | [0.46323407, 23.498016, 563.923, 56.076736, 8732.958]  |
+------+------------------------------------+--------------------------------------------------------+
4 rows in set (0.00 sec)

mysql> select * from vtab64;
+------+-------------------------------------+-------------------------------------------------------------------+
| id   | vecf64_3                            | vecf64_5                                                          |
+------+-------------------------------------+-------------------------------------------------------------------+
|    1 | NULL                                | NULL                                                              |
|    2 | [0.8166459, 0.66616553, 0.4886152]  | NULL                                                              |
|    3 | [8.5606893, 6.7903588, 821.977768]  | [0.46323407, 23.49801546, 563.9229458, 56.07673508, 8732.9583881] |
|    4 | [0.9260021, 0.26637346, 0.06567037] | [0.45756745, 65.2996871, 321.623636, 3.60082066, 87.58445764]     |
+------+-------------------------------------+-------------------------------------------------------------------+
4 rows in set (0.00 sec)

mysql> select * from vtab64 join vtab32 on vtab64.vecf32_3 = vtab32.vecf32_3;
ERROR 20301 (HY000): invalid input: column 'vtab64.vecf32_3' does not exist
mysql> select * from vtab32 left join vtab64 on vtab64.vecf32_5 = vtab32.vecf32_5;
ERROR 20301 (HY000): invalid input: column 'vtab64.vecf32_5' does not exist
mysql> select id,vecf32_3 + vecf32_3, vecf32_5 + vecf32_5 from vtab32;
+------+-----------------------------------+---------------------------------------------------------+
| id   | vecf32_3 + vecf32_3               | vecf32_5 + vecf32_5                                     |
+------+-----------------------------------+---------------------------------------------------------+
|    1 | NULL                              | NULL                                                    |
|    2 | [1.6332918, 1.3323311, 0.9772304] | NULL                                                    |
|    3 | [0.3452598, 6.5817714, 60.866188] | [0.9010489, 4.3969054, 19.159504, 246.96078, 9271.788]  |
|    4 | [17.121378, 13.580718, 1643.9556] | [0.92646813, 46.996033, 1127.846, 112.15347, 17465.916] |
+------+-----------------------------------+---------------------------------------------------------+
4 rows in set (0.01 sec)
